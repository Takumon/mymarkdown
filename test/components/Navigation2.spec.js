import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import {
  MdIcon,
  MdButton,
  MdTabs,
} from 'vue-material/dist/components'

import TestTargetVue from '../../components/Navigation2.vue'

jest.useFakeTimers();

describe('Navigation2.vue', () => {
  let wrapper
  let store

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(MdIcon)
    localVue.use(MdButton)
    localVue.use(MdTabs)
    localVue.use(Vuex)


    store = new Vuex.Store({
      state: {
        textEditorPreviewMode: 'tab-editor-and-preview',
      },
      actions: {
        setTextEditorPreviewMode ({ commit }, mode) {
          return new Promise((resolve, reject) => {
            commit( 'setTextEditorPreviewMode', { mode } )
            resolve()
          })
        },
      },
      mutations: {
        setTextEditorPreviewMode (state, payload) {
          state.textEditorPreviewMode = payload.mode
        },
      }
    })

    wrapper = shallowMount(TestTargetVue, {
      store,
      localVue,
    })
    jest.runAllTimers()
  })

  describe('初期表示', () => {

    test('スナップショット', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('編集かつプレビューモードが有効', () => {
      expect(store.state.textEditorPreviewMode).toEqual('tab-editor-and-preview')

      const $editorAndPrevieweTabButton = wrapper.find('[mdicon="chrome_reader_mode"]')
      expect($editorAndPrevieweTabButton.classes()).toContain('md-active')
    })
  })

  describe('プレビュータブクリック時', () => {
    beforeEach(() => {
      const $previewTabButton = wrapper.find('[mdicon="description"]')
      $previewTabButton.trigger('click')
    })

    test('プレビューモードが有効', () => {
      const $previewTabButton = wrapper.find('[mdicon="description"]')
      expect($previewTabButton.classes()).toContain('md-active')
      expect(store.state.textEditorPreviewMode).toEqual('tab-preview')
    })
  })

  describe('エディタータブクリック時', () => {
    beforeEach(() => {
      const $editorTabButton = wrapper.find('[mdicon="mode_edit"]')
      $editorTabButton.trigger('click')
    })

    test('エディターモードが有効', () => {
      const $editorTabButton = wrapper.find('[mdicon="mode_edit"]')
      expect($editorTabButton.classes()).toContain('md-active')
      expect(store.state.textEditorPreviewMode).toEqual('tab-editor')
    })
  })

})
