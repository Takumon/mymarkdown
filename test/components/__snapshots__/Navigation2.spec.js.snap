// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation2.vue 初期表示 スナップショット 1`] = `
<div class="md-toolbar-row">
  <div class="md-tabs md-primary md-alignment-left md-no-transition md-theme-default">
    <div class="md-tabs-navigation md-elevation-0">
      <button type="button" class="md-button md-theme-default" mdicon="mode_edit">
        <div class="md-ripple">
          <div class="md-button-content">
            <i class="md-icon md-icon-font md-tab-icon md-theme-default">mode_edit</i> <span class="md-tab-label"></span></div>
        </div>
      </button>
      <button type="button" class="md-button md-theme-default md-active" mdicon="chrome_reader_mode">
        <div class="md-ripple">
          <div class="md-button-content">
            <i class="md-icon md-icon-font md-tab-icon md-theme-default">chrome_reader_mode</i> <span class="md-tab-label"></span></div>
        </div>
      </button>
      <button type="button" class="md-button md-theme-default" mdicon="description">
        <div class="md-ripple">
          <div class="md-button-content">
            <i class="md-icon md-icon-font md-tab-icon md-theme-default">description</i> <span class="md-tab-label"></span></div>
        </div>
      </button> <span class="md-tabs-indicator"></span></div>
    <div class="md-content md-tabs-content md-theme-default" style="height: 0px; display: none;">
      <div class="md-tabs-container">
        <div id="tab-editor" class="md-tab"></div>
        <div id="tab-editor-and-preview" class="md-tab"></div>
        <div id="tab-preview" class="md-tab"></div>
      </div>
    </div>
  </div>
</div>
`;
